### PR TITLE
Users/pragupta/azurepwrshelltask

### PR DESCRIPTION
**Pull request by pragupta-ms**

### CODE REVIEW

File: Tasks/AzurePowerShell/AzurePowerShell.ps1 [hunk 1]
```diff
@@ -20,6 +24,14 @@ Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
 Initialize-Azure
 
 # Trace the expression as it will be invoked.
+If ($scriptType -eq "InlineScript") {
+    $tempFileName = [guid]::NewGuid().ToString() + ".ps1";
+    $scriptPath = [System.IO.Path]::Combine([System.IO.Path]::GetTempPath(), $tempFileName);
+    ($scriptInline | Out-File $scriptPath)
+
+    Write-Host "tempFile= $scriptPath"
```
Comment (Ajay-MS) [inline Tasks/AzurePowerShell/AzurePowerShell.ps1]: Add localized string here.

File: Tasks/AzurePowerShell/AzurePowerShell.ps1 [hunk 2]
```diff
@@ -2,13 +2,17 @@ Trace-VstsEnteringInvocation $MyInvocation
 Import-VstsLocStrings "$PSScriptRoot\Task.json"
 
 # Get inputs.
-$scriptPath = Get-VstsInput -Name ScriptPath -Require
+$scriptType = Get-VstsInput -Name ScriptType -Require
```
Comment (ericsciple) [inline Tasks/AzurePowerShell/AzurePowerShell.ps1]: when script type is filepath, then shouldn't ScriptPath be required? i.e. `Get-VstsInput -Name ScriptPath -Require`

similar thing for inline scenario

File: Tasks/AzurePowerShell/AzurePowerShell.ps1 (hunk 1 above)
Comment (ericsciple) [inline Tasks/AzurePowerShell/AzurePowerShell.ps1]: this looks like debug messaging to me. consider the write-verbose instead.

File: Tasks/AzurePowerShell/task.json [hunk 3]
```diff
@@ -55,12 +55,39 @@
...
         },
         {
+            "name": "ScriptType",
+            "type": "pickList",
+            "label": "Script Type",
+            "required": true,
+            "helpMarkDown": "Type of the script: File Path or Inline Script",
+            "defaultValue": "FilePath",
+            "options": {
+                "FilePath": "AzurePowerShell Script File Path",
```
Comment (ericsciple) [inline Tasks/AzurePowerShell/task.json]: ...nit continued :)

also, i would even say having "AzurePowerShell" in the dropdown item text may be confusing. the task initializes the Azure module and then runs the PowerShell script. other than the name of the task, there is no such thing as "Azure PowerShell"

**Comment by ericsciple** [comment]
i believe @kmkumaran 's team owns this task. i'm happy to provide a review, but signoff should come from the owners of this task.

**Comment by ericsciple** [comment]
nit, also i noticed the commits contains a merge commit. rebase is nicer for viewing history. sometimes merges are convenient, but for a small changes generally rebase is better

**Comment by kmkumaran** [comment]
@ericsciple - sure. My team will own the final sign-off for this task.

**Pull request merged**